### PR TITLE
HTCONDOR-1527 Improve vault token weakening

### PR DESCRIPTION
**Pull request by DrDaveD**
Addresses [HTCONDOR-1527](https://opensciencegrid.atlassian.net/browse/HTCONDOR-1527).  Also includes several other small miscellaneous improvements in the same two vault-integration files that users have run into.  Each of those improvements is in a separate commit. 

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

[HTCONDOR-1527]: https://opensciencegrid.atlassian.net/browse/HTCONDOR-1527?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jasoncpatton** [comment]
(Edited OP to fix checkboxes.)

File: src/condor_credd/condor_credmon_oauth/condor_vault_storer [hunk 1]
```diff
@@ -72,25 +72,70 @@ if [ "" != "$VERBOSE" ]; then
...
 
+# make sure we can find condor_store_cred
+PATH=/usr/sbin:$PATH
+
 # Keep the standard duration vault token in $VTOKEN-$SERVICE to make
 #  sure that credmon has a long-duration one, but copy it to $VTOKEN if a
 #  new one is generated.
-ID="`id -u`"
-VTOKEN="/tmp/vt_u$ID"
+ID="${CONDOR_VAULT_STORER_ID:-u`id -u`}"
```
[the remark this answers is not in the document]
Comment (DrDaveD) [reply src/condor_credd/condor_credmon_oauth/condor_vault_storer:86]: Actually it is used for a service that needs to impersonate multiple users.  I will add a commit that explains that and also adds another similar variable CONDOR_VAULT_STORER_USER which enables passing a different user name to condor_store_cred.

File: src/condor_credd/condor_credmon_oauth/condor_vault_storer [hunk 2]
```diff
@@ -72,25 +72,70 @@ if [ "" != "$VERBOSE" ]; then
...
 NL="
 "
 
+# These two functions depend on some variables that are set in the loop below,
+# but they do not need to be defined multiple times so they are outside of the
+# loop.
+querycred() {
+    STORENAME="$1"
+    verbose "Checking if $@ credentials exist"
+    STOREMSG="`condor_store_cred query-oauth -s "$@" >$STOREOUT 2>&1`"
```
[the remark this answers is not in the document]
Comment (DrDaveD) [reply src/condor_credd/condor_credmon_oauth/condor_vault_storer:103]: Yeah that's a documented bash feature.  Putting double quotes around an array referenced like that makes it quote each individual element, and `"$@"` is a shorthand for expanding the function (or script) parameters as an array.

File: src/condor_credd/condor_credmon_oauth/condor_vault_storer [hunk 3]
```diff
@@ -115,100 +160,159 @@ for REQUEST; do
...
+            # Copy bearer token to $BTOKEN atomically
+            TMPFILE="`mktemp $BTOKEN.XXXXXXXXXX`"
+            cat $BTOKEN-$SERVICE >$TMPFILE
+            mv $TMPFILE $BTOKEN
+        fi
+
+        STOREVAULTTOKEN=false
+        VAULTURL=""
+        CREDLINES="`echo "$CRED"|wc -l`"
+        case $CREDLINES in
```
[the remark this answers is not in the document]
Comment (DrDaveD) [reply src/condor_credd/condor_credmon_oauth/condor_vault_storer:227]: That's true.  In order to help prevent the output from changing, however, the htgettoken man page documents this behavior.

**Review by jasoncpatton — APPROVED** [review]
Additional commits approved.

**Pull request merged**